dds/idl/marshal_generator.cpp

```diff
@@ -280,36 +280,6 @@ namespace {
     return s.substr(i, cxx.size()) == cxx;
   }
 
-  std::string valid_var_name(const std::string& str)
-  {
-    // Replace invalid characters with a single underscore
-    const std::string invalid_chars("<>()[]*.: ");
-    std::string s;
-    char last_char = '\0';
-    for (size_t i = 0; i < str.size(); ++i) {
-      const char c = str[i];
-      if (invalid_chars.find(c) == std::string::npos) {
-        s.push_back(c);
-        last_char = c;
-      } else if (last_char != '_') {
-        s.push_back('_');
-        last_char = '_';
-      }
-    }
-
-    // Remove underscores at start and end
-    if (s.size() > 1 && s[0] == '_') {
-      if (!cxx_escape(s, 0)) {
-        s.erase(0, 1);
-      }
-    }
-    if (s.size() > 1 && s[s.size() - 1] == '_') {
-      s.erase(s.size() - 1, 1);
-    }
```

Comment:
It looks like you left this part out, why?

Reply:
if uuid and uuid_ were provided, it was causing them both to become uuid which lead to collision.

Reply:
Was an earlier version of the fix to allow underscored names but didn't resolve collision issue. Looks like it can be added back now.